deepmd/infer/model_devi.py

```diff
@@ -208,17 +211,21 @@ def make_model_devi(
     for system in all_sys:
         # create data-system
         dp_data = DeepmdData(system, set_prefix, shuffle_test=False, type_map=tmap)
+        mixed_type = dp_data.mixed_type
 
         data_sets = [dp_data._load_set(set_name) for set_name in dp_data.dirs]
         nframes_tot = 0
         devis = []
         for data in data_sets:
             coord = data["coord"]
             box = data["box"]
-            atype = data["type"][0]
+            if mixed_type:
+                atype = data["type"]
+            else:
+                atype = data["type"][0]
             if not dp_data.pbc:
                 box = None
-            devi = calc_model_devi(coord, box, atype, dp_models)
+            devi = calc_model_devi(coord, box, atype, dp_models, mixed_type=mixed_type)
```

Comment:
It seems to me these lines that you fixed are not covered

![Screenshot from 2023-03-29 14-02-51](https://user-images.githubusercontent.com/9496702/228628073-ad25d70e-7ad3-46ac-89eb-fe12c6b7e45f.png)